src/invidious/helpers/youtube_api.cr

```diff
@@ -2,120 +2,398 @@
 # This file contains youtube API wrappers
 #
 
-# Hard-coded constants required by the API
-HARDCODED_API_KEY     = "AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8"
-HARDCODED_CLIENT_VERS = "2.20210330.08.00"
+module YoutubeAPI
+  extend self
 
-####################################################################
-# make_youtube_api_context(region)
-#
-# Return, as a Hash, the "context" data required to request the
-# youtube API endpoints.
-#
-def make_youtube_api_context(region : String | Nil) : Hash
-  return {
-    "client" => {
-      "hl"            => "en",
-      "gl"            => region || "US", # Can't be empty!
-      "clientName"    => "WEB",
-      "clientVersion" => HARDCODED_CLIENT_VERS,
+  # Enumerate used to select one of the clients supported by the API
+  enum ClientType
+    Web      = 0
+    Android  = 1
+    WebEmbed = 2
+  end
+
+  # List of hard-coded values used by the different clients
+  HARDCODED_CLIENTS = {
+    ClientType::Web => {
+      name:    "WEB",
+      version: "2.20210721.00.00",
+      api_key: "AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8",
+    },
+    ClientType::Android => {
+      name:    "ANDROID",
+      version: "16.20.35",
+      api_key: "AIzaSyA8eiZmM1FaDVjRy-df2KTyQ_vz_yYM39w",
+    },
+    ClientType::WebEmbed => {
+      name:    "WEB_EMBEDDED_PLAYER", # 56
+      version: "1.20210721.1.0",
```

Comment:
The same client version can be used for both the web embed and the normal web client

Reply:
I just checked, and they're indeed different:

```json
...
"GAPI_LOCALE":"en_US",
"GL":"FR",
"HL":"en",
"HTML_LANG":"en",
"INNERTUBE_API_KEY":"AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8",
"INNERTUBE_API_VERSION":"v1",
"INNERTUBE_CLIENT_NAME":"WEB_EMBEDDED_PLAYER",
"INNERTUBE_CLIENT_VERSION":"1.20210721.1.0",
"INNERTUBE_CONTEXT": {}
...
```

Reply:
Yep, seems like you're right there; [I was thinking of the client parameters used for bypassing age restriction instead of the `WEB_EMBEDDED_PLAYER`.](https://github.com/iv-org/invidious/issues/2189#issuecomment-887253427)

And speaking of the bypass, we'll still some way to specify the `ClientScreen` parameter with the normal web client. My suggestion would be to add another parameter in `ClientConfig`. 

Something along the lines of:
```crystal
  struct ClientConfig
    ...
    # Additional context information
    # This is then merged (#merge!) with the usual context created from `make_context`
    property additional_context : Hash(String, String)?

    # Initialization function
    def initialize(
      *,
      @client_type = ClientType::Web,
      @region = "US",
      @proxy_region = nil,
      @additional_context=nil,
    )
```

```crystal
private def make_context(client_config : ClientConfig | Nil) : Hash
  ...
  if client_config.additional_context
    client_context.merge!(client_config.additional_context)
  end

  return client_context
end
```